pkg/insights/insightsclient/requests.go

```diff
@@ -12,21 +12,22 @@ import (
 	"k8s.io/klog/v2"
 
 	"github.com/openshift/insights-operator/pkg/authorizer"
+	"github.com/openshift/insights-operator/pkg/insights"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
-func (c *Client) SendAndGetID(ctx context.Context, endpoint string, source Source) (string, error) {
+func (c *Client) SendAndGetID(ctx context.Context, endpoint string, source Source) (string, int, error) {
```

Comment:
I'm having some concerns about confusing use of second return parameter. This is part of the same nit with upcoming comments. I prefer to leave the comments inline to avoid more confusion :D

As a summary, I'd assign a variable to default returning code, to improve readability. Then later use the same codes as parameters for a control function, moving the transformation of the data to that function itself. That way code is more straightforward IMHO, and avoids confusion on why transforming the data, why sending a 0 as a string, 0 int returnings with no context, etc...

I think it's a simple change but that improves readability and avoids any confusion on why or what the data is used for.

Please, check upcoming messages :point_down: